### ISSUE TITLE
[Bug] [Android] HTTP call is crashing application

### ISSUE THREAD

**Issue by mauricio-codecraft** [silkimen/cordova-plugin-advanced-http#274]
**Describe the bug**
HTTP call is crashing the application while running (http call) on devapp. Not sure but it seems to be related to OKHTTP_VERSION.

**System info**

Ionic:

   Ionic CLI                     : 5.4.7 (/home/mauricio/.nvm/versions/node/v12.13.0/lib/node_modules/ionic)
   Ionic Framework               : @ionic/angular 4.11.5
   @angular-devkit/build-angular : 0.801.3
   @angular-devkit/schematics    : 8.1.3
   @angular/cli                  : 8.1.3
   @ionic/angular-toolkit        : 2.1.1

Cordova:

   Cordova CLI       : 9.0.0 (cordova-lib@9.0.1)
   Cordova Platforms : android 8.1.0
   Cordova Plugins   : cordova-plugin-ionic-keyboard 2.2.0, cordova-plugin-ionic-webview 4.1.3, (and 6 other plugins)

Utility:

   cordova-res : 0.8.1 
   native-run  : not installed

System:

   NodeJS : v12.13.0 (/home/mauricio/.nvm/versions/node/v12.13.0/bin/node)
   npm    : 6.12.0
   OS     : Linux 4.15

 "cordova-android": "^8.1.0",
 "cordova-plugin-advanced-http": "^2.2.0",
 "cordova-plugin-file": "^6.0.2",
 affected device(s): [Android 9]
 
"dependencies": {
    "@angular/common": "~8.1.2",
    "@angular/core": "~8.1.2",
    "@angular/forms": "~8.1.2",
    "@angular/platform-browser": "~8.1.2",
    "@angular/platform-browser-dynamic": "~8.1.2",
    "@angular/router": "~8.1.2",
    "@ionic-native/core": "^5.0.0",
    "@ionic-native/http": "^5.17.1",
    "@ionic-native/splash-screen": "^5.0.0",
    "@ionic-native/status-bar": "^5.0.0",
    "@ionic/angular": "^4.7.1",
    "cordova-android": "^8.1.0",
    "cordova-plugin-advanced-http": "^2.2.0",
    "cordova-plugin-file": "^6.0.2",
    "core-js": "^2.5.4",
    "rxjs": "~6.5.1",
    "tslib": "^1.9.0",
    "zone.js": "~0.9.1"
  },
  "devDependencies": {
    "@angular-devkit/architect": "~0.801.2",
    "@angular-devkit/build-angular": "~0.801.2",
    "@angular-devkit/core": "~8.1.2",
    "@angular-devkit/schematics": "~8.1.2",
    "@angular/cli": "~8.1.2",
    "@angular/compiler": "~8.1.2",
    "@angular/compiler-cli": "~8.1.2",
    "@angular/language-service": "~8.1.2",
    "@ionic/angular-toolkit": "^2.1.1",
    "@types/jasmine": "~3.3.8",
    "@types/jasminewd2": "~2.0.3",
    "@types/node": "~8.9.4",
    "codelyzer": "^5.0.0",
    "cordova-plugin-device": "^2.0.2",
    "cordova-plugin-ionic-keyboard": "^2.2.0",
    "cordova-plugin-ionic-webview": "^4.1.3",
    "cordova-plugin-splashscreen": "^5.0.2",
    "cordova-plugin-statusbar": "^2.4.2",
    "cordova-plugin-whitelist": "^1.3.3",
    "jasmine-core": "~3.4.0",
    "jasmine-spec-reporter": "~4.2.1",
    "karma": "~4.1.0",
    "karma-chrome-launcher": "~2.2.0",
    "karma-coverage-istanbul-reporter": "~2.0.1",
    "karma-jasmine": "~2.0.1",
    "karma-jasmine-html-reporter": "^1.4.0",
    "protractor": "~5.4.0",
    "ts-node": "~7.0.0",
    "tslint": "~5.15.0",
    "typescript": "~3.4.3"
  },
  "description": "An Ionic project",
  "cordova": {
    "plugins": {
      "cordova-plugin-whitelist": {},
      "cordova-plugin-statusbar": {},
      "cordova-plugin-device": {},
      "cordova-plugin-splashscreen": {},
      "cordova-plugin-ionic-webview": {
        "ANDROID_SUPPORT_ANNOTATIONS_VERSION": "27.+"
      },
      "cordova-plugin-ionic-keyboard": {},
      "cordova-plugin-advanced-http": {
        "OKHTTP_VERSION": "3.10.0"
      }
    },
    "platforms": [
      "android"
    ]
  }

**Are you using ionic-native-wrapper?**
No.

**Minimum viable code to reproduce**

```
getStuff() {
    alert('before call!!')
    this.http.get('http://ionic.io', {}, {})
    .then(data => {
      alert(data.data)
      console.log(data.status);
      console.log(data.data); // data received by server
      console.log(data.headers);
    })
    .catch(error => {
      alert(error.erorr)
      console.log(error.status);
      console.log(error.error); // error message as string
      console.log(error.headers);
    });
    alert('after call!!')
  }
```

**Run command**
ng run app:serve --host=0.0.0.0 --port=8100

**Comment by beruic** [comment]
Can you confirm that this only happens on debug builds?

I'm experiencing random crashes on sessions downloading a lot of files.

**Comment by arnoldgamarram** [comment]
Hi, use this commands, it work well for me.

cordova plugin add cordova-plugin-advanced-http --variable OKHTTP_VERSION="**3.12.1**"
 
or set it into project.properties directly (not recomended).

cordova.system.library.4=com.squareup.okhttp3:okhttp-urlconnection:**3.12.1**

**Comment by arnoldgamarram** [comment]
maybe you can share the code on github and paste the url here.

**Comment by ryaa** [comment]
I use capacitor and --variable does not work there so I manually changed build.gradle by adding the below under allprojects so it looks as below

```
allprojects {
    repositories {
        google()
        jcenter()
    }
    configurations.all {
        resolutionStrategy.dependencySubstitution {
            // Substitute one module dependency for another
            substitute module('com.squareup.okhttp3:okhttp-urlconnection:3.10.0') with module('com.squareup.okhttp3:okhttp-urlconnection:3.12.1')
        }
    }
}
```

**Comment by silkimen** [comment]
Should be fixed #297. Closing this, feel free to open a new issue, if this won't solve your problem! 👍
